addons/mail/models/mail_alias_domain.py

```diff
@@ -180,17 +180,43 @@ def _sanitize_configuration(self, config_values):
     @api.model
     def _find_aliases(self, email_list):
         """ Utility method to find both alias domains aliases (bounce, catchall
-        or default from) and mail aliases from an email list. """
+        or default from) and mail aliases from an email list.
+
+        :param email_list: list of normalized emails; normalization / removing
+            wrong emails is considered as being caller's job
+        """
         if not email_list:
             return email_list
+
+        catchall_domains_allowed = list(filter(None, (self.env["ir.config_parameter"].sudo().get_param(
+            "mail.catchall.domain.allowed") or '').split(',')))
+        if catchall_domains_allowed:
+            catchall_domains_allowed += self.env['mail.alias.domain'].search([]).mapped('name')
```

Comment:
Note: done below, useless search, instead put below code up

Reply:
Good catch 👍 

Went through it again, seems legit and well-covered
Thanks👌